llvm/lib/Target/EVM/EVMConstantUnfolding.cpp

```diff
@@ -75,9 +76,71 @@ static cl::opt<unsigned> MetadataSize("evm-metadata-size", cl::Hidden,
                                       cl::init(0),
                                       cl::desc("EVM metadata size"));
 
+static cl::opt<unsigned> ConstantSpillThreshold(
```

Comment:
Could you explain why we need to couple spilling and materialization logic? How one affects the other?